src/Core/Domain/Profile/Employee/Exception/UndefinedEmployeeStatusException.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Profile\Employee\Exception;
+
+/**
+ * Class UndefinedEmployeeAccountStatusException.
+ */
+class UndefinedEmployeeStatusException extends EmployeeException
```

Comment:
i dont know.. i dont want to throw single generic exception (e.g. `EmployeeException`) for different errors, do you have other ideas? :thinking:

Reply:
Sometimes, you use exception codes as php constants. Looks good IMO.

Reply:
yup, when constraint exception happens. Since its pretty generic, we use codes to specify which constraint was violated. do you suggest having just one `EmployeeException` with a lot of codes?

Reply:
No you're right 😄 thank you